lightning/src/ln/channel.rs

```diff
@@ -1684,63 +1684,108 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 		(COMMITMENT_TX_BASE_WEIGHT + num_htlcs as u64 * COMMITMENT_TX_WEIGHT_PER_HTLC) * self.feerate_per_kw as u64 / 1000 * 1000
 	}
 
-	// Get the commitment tx fee for the local (i.e our) next commitment transaction
-	// based on the number of pending HTLCs that are on track to be in our next
-	// commitment tx. `addl_htcs` is an optional parameter allowing the caller
-	// to add a number of additional HTLCs to the calculation. Note that dust
-	// HTLCs are excluded.
-	fn next_local_commit_tx_fee_msat(&self, addl_htlcs: usize) -> u64 {
+	// Get the commitment tx fee for the local's (i.e. our) next commitment transaction based on the
+	// number of pending HTLCs that are on track to be in our next commitment tx, plus an additional
+	// HTLC if `fee_spike_buffer_htlc` is Some, plus a new HTLC given by `new_htlc_amount`. Dust HTLCs
+	// are excluded.
+	fn next_local_commit_tx_fee_msat(&self, new_htlc_amount: u64, locally_offered: bool, fee_spike_buffer_htlc: Option<()>) -> u64 {
```

Comment:
I'm not a huge fan of passing bools to a function as it isn't obvious at the call site as to what the argument means. Consider using an enum, possibly combining it with the amount from the first parameter such that the combined parameter represents the HTLC with some value.

Reply:
Sure, would you be fine with:
```
enum HTLCInitiator {
    LocallyOffered
    RemoteOffered
}
struct HTLCCandidate {
    amount_msat: u64,
    origin: HTLCInitiator
}
fn next_local_commit_tx_fee_msat(&self, htlc: HTLCCandidate....) { ... }
```

Reply:
Maybe `LocalOffered` for consistency, not withstanding the glossary's usage of holder/counterparty (or perhaps operator/counterparty in the future). Probably best to stick to the terminology already in use here.

If call sites seem verbose, consider adding `local` and `remote` constructors, or something similarly named.

Reply:
I think that turned out much nicer, lmk what you think.

Reply:
Looks great! Thanks or making the change.